titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/PodToTaskMapper.java

```diff
@@ -241,11 +246,43 @@ private Either<TaskStatus, String> handlePodRunning() {
         if (isBefore(Started, taskState)) {
             return unexpected("pod state (Running) not consistent with the task state");
         }
+
+        long now = titusRuntime.getClock().wallTime();
+        if ("NodeLost".equals(podWrapper.getReason())) {
+            // If task is still in the Accepted state, move it first, as we know it was scheduled.
+            if (taskState == TaskState.Accepted) {
+                return Either.ofValue(TaskStatus.newBuilder()
+                        .withState(Launched)
+                        .withReasonCode(REASON_NORMAL)
+                        .withReasonMessage(String.format(
+                                "The pod is scheduled but the communication with its node is lost. If not recovered in %s, the task will be marked as failed",
+                                DateTimeExt.toTimeUnitString(configuration.getNodeLostTimeoutMs())
+                        ))
+                        .withTimestamp(now)
+                        .build()
+                );
+            }
+            // Give it a chance to recover from the `NodeLost` state.
+            long lastUpdateTime = task.getStatus().getTimestamp();
+            long deadline = lastUpdateTime + configuration.getNodeLostTimeoutMs();
+            if (deadline > now) {
+                // Keep the existing state
+                return Either.ofValue(task.getStatus());
+            }
+            // We waited long enough. Time to mark this task as failed.
+            return Either.ofValue(TaskStatus.newBuilder()
+                    .withState(Finished)
+                    .withReasonCode(REASON_TASK_LOST)
+                    .withReasonMessage("The node where task was scheduled is lost")
```

Comment:
`The node where task was scheduled is lost before it got a chance to start`
(or something that indicates that it was only on the Accepted state. Right?

Reply:
Actually, this logic is executed when the pod is in Running state